packages/discord-utilities/src/lib/regexes.ts

```diff
@@ -0,0 +1,114 @@
+ * @raw `/^(?:<#)?(\d{17,19})>?$/`
+ * @remark Capture group 1 is the ID of the channel. It is named `id`.
+ */
+export const ChannelMentionRegex = /^(?:<#)?(?<id>\d{17,19})>?$/;
+
+/**
+ * Regex that matches links on the known Discord host names
+ * @raw `/^(?:\w+\.)?(?:discordapp.com|discord.gg|discord.com)$/i`
+ * @remark The regex is case insensitive
+ */
+export const DiscordHostnameRegex = /^(?:\w+\.)?(?:discordapp\.com|discord\.gg|discord\.com)$/i;
+
+/**
+ * Regex that can can capture the code of Discord invite links
+ * @raw `^(?:https?:\/\/)?(?:www.)?(?:discord\.gg\/|discordapp\.com\/invite\/)?(?<code>[\w\d-]{2,})$`
+ * @remark Capture group 1 is the invite URL's unique code. It is named `code`.
+ */
+export const DiscordInviteLinkRegex = /^(?:https?:\/\/)?(?:www\.)?(?:discord\.gg\/|discordapp\.com\/invite\/)?(?<code>[\w\d-]{2,})$/i;
+
+/**
+ * Regex that can capture the ID of any animated or non-animated custom Discord emoji
+ * @raw `/^(?:<a?:\w{2,32}:)?(\d{17,21})>?$/`
+ * @remark Capture group 1 is the ID of the emoji. It is named `id`.
+ */
+export const EmojiRegex = /^(?:<a?:\w{2,32}:)?(?<id>\d{17,21})>?$/;
+
+/**
+ * Regex that matches any animated or non-animated custom Discord emoji.
+ * Unlike [[EmojiRegex]] It can be a substring of a larger string.
+ * @raw `/<a?:\w{2,32}:\d{17,18}>/`
+ */
+export const FormattedCustomEmoji = /<a?:\w{2,32}:\d{17,18}>/;
+
+/**
+ * Regex that can capture any animated or non-animated custom Discord emoji.
+ * Similar to [[FormattedCustomEmoji]] and unlike [[EmojiRegex]] can also be a substring of a larger string.
+ * @raw `/^(a?):([^:]+):(\d{17,19})$/`
+ * @remark Capture group 1 can be used to determine whether the emoji is animated or not. It is named `animated`.
+ * @remark Capture group 2 is the name of the emoji as it is typed in a message. It is named `name`.
+ * @remark Capture group 2 is the ID of the emoji. It is named `id`.
+ */
+export const FormattedCustomEmojiWithGroups = /^(?<animated>a?):(?<name>[^:]+):(?<id>\d{17,19})$/;
+
+/**
+ * Regex that matches any URL starting with `http` or `https`
+ * @raw `/^https?:\/\//`
+ * @remark for WebSocket URLs see [[WebsocketGenericUrlRegex]]
+ */
+export const HttpUrlRegex = /^https?:\/\//;
+
+/**
+ * Regex that can capture the Guild, Channel, and Message ID based on a shareable Discord message link.
+ * @raw `/^(?:https:\/\/)?(?:ptb\.|canary\.)?discord(?:app)?\.com\/channels\/(?<guildId>\d{17,19})\/(?<channelId>\d{17,19})\/(?<messageId>\d{17,19})$/`
+ * @remark Capture group 1 is the ID of the guild the message was sent in. It is named `guildId`.
+ * @remark Capture group 2 is the ID of the channel in that guild the message was sent in. It is named `channelId`.
+ * @remark Capture group 3 is the ID of the message itself. It is named `messageId`.
+ */
+export const MessageLinkRegex = /^(?:https:\/\/)?(?:ptb\.|canary\.)?discord(?:app)?\.com\/channels\/(?<guildId>\d{17,19})\/(?<channelId>\d{17,19})\/(?<messageId>\d{17,19})$/;
```

Comment:
First ID can also be `@me`... should add that to the regex

Reply:
Fixed